src/Sulu/Component/Content/Security/Listener/PermissionUpdateListener.php

```diff
@@ -17,6 +17,8 @@
 
 /**
  * This class listens on permission updates, and set the permission on the document described in the event.
+ *
+ * @deprecated
```

Comment:
We should probably think about removing the new deprecated classes here immediately. Not sure if it is an acceptable BC break.